### PR TITLE
Part 12 of unit tests update for KRaft

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This is the 12th and hopefully last part of the unit test refactoring to better support KRaft. It handles the `KafkaClusterWithKRaftTest` class. Most of the tests inside are slightly different duplicates of tests in `KafkaClusterTest` and `KafkaClusterListenersTest`. So they are deleted or merged into these two classes. Also, some additional listener related tests were moved from `KafkaClusterTest` to `KafkaClusterListenersTest`.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java [hunk 1]
```diff
@@ -4097,4 +4116,77 @@ public void testCustomizedPodSetInNodePool()   {
...
+    public void testNodesAndStatuses()  {
+        //KafkaCluster KC = KafkaCluster.fromCrd(
+        //        Reconciliation.DUMMY_RECONCILIATION,
+        //        KAFKA,
+        //        List.of(KAFKA_POOL_CONTROLLERS, KAFKA_POOL_BROKERS),
+        //        VERSIONS,
+        //        KafkaVersionTestUtils.DEFAULT_KRAFT_VERSION_CHANGE,
+        //        KafkaMetadataConfigurationState.KRAFT,
+        //        null, SHARED_ENV_PROVIDER
+        //);
```
Comment (im-konge) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java:4142]: Is this needed?

**Review by im-konge — APPROVED** [review]
Just one thing, thanks.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java]: Good catch, fixed. Thanks.

**Review by ppatierno — APPROVED** [review]
LGTM

**Pull request merged**